pkg/cloud/services/ec2/instances.go

```diff
@@ -173,7 +173,7 @@ func (s *Service) CreateInstance(scope *scope.MachineScope, userData []byte) (*i
 	}
 	input.SubnetID = subnetID
 
-	if !scope.IsEKSManaged() && s.scope.Network().APIServerELB.DNSName == "" {
+	if !scope.IsExternallyManaged() && !scope.IsEKSManaged() && s.scope.Network().APIServerELB.DNSName == "" {
```

Comment:
I was under the impression that it being externally managed means there should be no reconciliation at all? So do we not need something like
```
if scope.IsExternallyManaged() {
  // Should not reconcile
  return nil, nil
}
```

Reply:
This change is related to instance creation, we are skipping logic that is relying on infrastructure created by the managed cluster.

Reply:
Does this mean that if we are using external but still provide the APIServer DNSName then we still work as expected?

Reply:
this logic is only used to check that a load balancer was created.